Benjamin - Benjamin-Tu/bt_week4_assignment.do

```diff
@@ -0,0 +1,188 @@
+
+
+global wd "/Users/benjamintu/Downloads/Econ490_Fall2022/Week4/04_assignment/data"
+
+use "$wd/village_pixel.dta", clear
+
+* Q1 a)
+egen pixel_mean = mean(payout),by(pixel) //calculate the mean for each different pixel.
+
+gen pixel_consistent = 0 if pixel_mean== 0 | pixel_mean ==1    // pixel is consistent if the mean is an integer.
+
+replace pixel_consistent = 1 if pixel_mean!= 0 & pixel_mean!= 1 // if the mean is not an integer, pixel is not consistent
+
+* b)
+/*
+why the value changes when I try to destring?
+gen nu_pixel = substr(pixel,3,4)
+destring (nu_pixel), replace
+egen nu_pixel_mean = mean(nu_pixel),by(village)
+*/
+
+sort village
+
+egen tag_pixel_village = tag(village pixel) 
+
+egen tag_pixel_village_mean = sum(tag_pixel_village),by(village)
+
+gen pixel_village=0 if  tag_pixel_village_mean == 1
+
+replace pixel_village=1 if tag_pixel_village_mean >1
+
+* c)
+gen village_categories=1 if pixel_village==0 
+
+replace village_categories=2 if pixel_village==1 & pixel_consistent==0
+
+replace village_categories=3 if pixel_village==1 & pixel_consistent==1
+
+tab hhid if village_categories==2
+
+*Q2
+use "$wd/psle_student_raw.dta", clear
+
+	levelsof schoolcode, local (list) // assign a number to each school
+	local n = 0 
+	foreach i of local list {
```

Comment:
I had trouble with this loop where it only cleaned one webpage rather than all 138. I'm not sure exactly why the loop didn't run through all of them, but I wonder if it has to do with how you're using the macros in the list you define using `levelsof` above.

Reply:
The loop in Ben's code doesn't work since there is a space between local and (list). It should be local(list) which will assign all schoolcode into a local named "list".

Reply:
Ah, this is a very helpful explanation! Thank you, Roy!